specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,146 @@
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_MAIN = 1,
+
+  /// Indicates that web resource is requested from shared worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SHARED_WORKER = 2,
+
+  /// Indicates that web resource is requested from service worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER = 4,
+
+  /// Indicates that web resource is requested from any supported source.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL = 0XFFFFFFFF
+} COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE);")
+
+[uuid(1cc6a402-3724-4e4a-9099-8cf60f2f93a1), object, pointer_default(unique)]
+interface ICoreWebView2: IUnknown {
+  /// Extends AddWebResourceRequestedFilter to support events from shared
+  /// and service workers. To receive all raised events filters have to be added
+  /// before main page navigation.
+  /// Because service worker runs separately from main page its WebResourceRequested
+  /// will be raised for all webviews which are subscribed to
+  /// COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER request source.
+  /// It is recommended to enable WebResourceRequested filter for service workers
+  /// only on a main WebView to avoid handling the same WebResourceRequested
+  /// event multiple times.
+  ///
+  /// For more information about web resource requested filters, navigate to
+  /// AddWebResourceRequestedFilter.
+  HRESULT AddWebResourceRequestedFilterWithRequestSource(
+    [in] LPCWSTR const uri,
+    [in] COREWEBVIEW2_WEB_RESOURCE_CONTEXT const resourceContext,
+    [in] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE const requestSource);
+
+  /// Removes a matching WebResource filter that was previously added for the
+  /// `WebResourceRequested` event.  If the same filter was added multiple
+  /// times, then it must be removed as many times as it was added for the
+  /// removal to be effective.  Returns `E_INVALIDARG` for a filter that was
+  /// never added.
+  HRESULT RemoveWebResourceRequestedFilterWithRequestSource(
+      [in] LPCWSTR const uri,
+      [in] COREWEBVIEW2_WEB_RESOURCE_CONTEXT const resourceContext,
+      [in] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE const requestSource);
+ }
+
+/// Event args for the `WebResourceRequested` event.
+[uuid(572f38f9-c317-4b6c-8dd2-c8b894779bbb), object, pointer_default(unique)]
+interface ICoreWebView2WebResourceRequestedEventArgs : IUnknown {
+  /// The web resource request source.
+  [propget] HRESULT RequestSource(
+  [out, retval] COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE* requestSource);
+}
+```
+
+## .NET and WinRT
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    [flags]
+    enum CoreWebView2WebResourceRequestSource
+    {
+        Main = 0x00000001,
+        SharedWorker = 0x00000002,
+        ServiceWorker = 0x00000004,
+        All = 0xFFFFFFFF,
+    };
+
+    runtimeclass CoreWebView2
+    {
+        void AddWebResourceRequestedFilter(
```

Comment:
If you copy this out of the generated Microsoft.Web.WebView2.Core.IDL there should be a block with an interface name containing these two new methods? We should include that here to make it clear we aren't breaking compat

Reply:
Just wanted to remind you that implementation for new method will fix OOP iframes events for default value, for which WebResourceRequested is not sent in old method.  Let me know if we want to keep the old name `AddWebResourceRequestedFilter`, or have new API `AddWebResourceRequestedFilterWithRequestSource`.

Reply:
No complaint about the name. Just want to see something like the following around the new methods which you should see in the generated IDL

```
[interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_16")]
{
...
}
```